untrusted/urts/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+use mc_sgx_urts_sys::{
+    sgx_create_enclave_from_buffer_ex, sgx_destroy_enclave, sgx_enclave_id_t, sgx_status_t,
+};
+use std::ops::Deref;
+use std::{os::raw::c_int, ptr};
+
+#[derive(PartialEq, Debug)]
+pub enum Error {
+    // An error provided from the SGX SDK
+    SgxStatus(sgx_status_t),
+}
+
+/// Struct for interfacing with the SGX SDK.  This should be used directly in
+/// sgx calls `ecall_some_function(*enclave, ...)`.
+///
+/// Avoid storing the de-referenced instance of the enclave.  The de-referenced
+/// value of the enclave will result in failures to the SGX SDK after the
+/// enclave is dropped.
+#[derive(Debug, PartialEq)]
+pub struct Enclave {
+    // The enclave ID, assigned by the SGX interface
+    id: sgx_enclave_id_t,
+}
+
+/// Build an [Enclave] for use with SGX calls.
+#[derive(Default)]
+pub struct EnclaveBuilder {
+    // The bytes for the enclave.
+    bytes: Vec<u8>,
+
+    // `true` if the enclave should be created in debug mode
+    debug: bool,
+}
+
+impl EnclaveBuilder {
+    /// Returns an EnclaveBuilder for the provided signed enclave.
+    ///
+    /// # Arguments
+    ///
+    /// * `bytes` - The bytes representing the enclave file.  This should be a
+    ///     signed enclave.
+    pub fn new(bytes: &[u8]) -> EnclaveBuilder {
+        EnclaveBuilder {
+            bytes: bytes.into(),
+            debug: false,
+        }
+    }
+
+    /// Toggle debugging of the enclave on or off.  The default is off.
+    ///
+    /// # Arguments
+    ///
+    /// * `debug` - `true` to enable enclave debugging, `false` to disable it.
+    #[must_use]
+    pub fn debug(mut self, debug: bool) -> EnclaveBuilder {
+        self.debug = debug;
+        self
+    }
+
+    /// Create the enclave
+    ///
+    /// Will talk to the SGX SDK to create the enclave.  Once the enclave has
+    /// been created then calls into the enclave can be made by de-referencing
+    /// the enclave.
+    ///
+    /// See
+    /// <https://download.01.org/intel-sgx/sgx-dcap/1.13/linux/docs/Intel_SGX_Enclave_Common_Loader_API_Reference.pdf>
+    /// for error codes and their meaning.
+    pub fn create(mut self) -> Result<Enclave, Error> {
```

Comment:
Let's tag this `must_use`

```suggestion
    #[must_use]
    pub fn create(mut self) -> Result<Enclave, Error> {
```

Reply:
I initially added this but had to remove it.
```
➜  untrusted git:(feature/untrusted) ✗ cargo clippy
   Compiling mc-sgx-urts-sys-types v0.1.0 (/home/nick/git/sgx_2/untrusted/urts_sys_types)
   Compiling mc-sgx-urts-sys v0.1.0 (/home/nick/git/sgx_2/untrusted/urts_sys)
    Checking mc-sgx-urts v0.1.0 (/home/nick/git/sgx_2/untrusted/urts)
warning: this function has an empty `#[must_use]` attribute, but returns a type already marked as `#[must_use]`
  --> urts/src/lib.rs:72:5
   |
72 |     pub fn create(mut self) -> Result<Enclave, Error> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::double_must_use)]` on by default
   = help: either add some descriptive text or remove the attribute
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#double_must_use
```
`Result` types are marked must use, https://doc.rust-lang.org/std/result/#results-must-be-used.  So marking the method results in a warning about double marking.